lib/core/include/common.mk

```
@@ -159,20 +163,36 @@ regenerate:
 # list, but restores them if the generation fails.
 .PHONY: try-regenerate
 try-regenerate:
-	@$(MAKE) --no-print-directory regenerate || (echo 'Regenerate failed, restoring files...'; git restore $(shell git ls-files $(GENERATED_FILES)))
+	$(eval _GENERATED_FILE_COUNT := $(words $(GENERATED_FILES)))
+	$(eval _GIT_GENERATED_FILES := $(shell echo "$(GENERATED_FILES)" | xargs git ls-files --))
+	$(eval _GIT_GENERATED_FILE_COUNT := $(words $(_GIT_GENERATED_FILES)))
+	@if $(MAKE) --no-print-directory regenerate; then \
+		echo "$(_GENERATED_FILE_COUNT) generated file(s) regenerated"; \
+	else \
+		echo "Regenerate failed, restoring $(_GIT_GENERATED_FILE_COUNT) files..."; \
+		echo "$(_GIT_GENERATED_FILES)" | xargs git restore --; \
+	fi
 
 # verify-generated --- Removes and regenerates all files in the GENERATED_FILES
 # list and checks for differences to the committed files. The target fails if
 # differences are detected.
 .PHONY: verify-generated
 verify-generated:
-	@PATH="$(PATH)" verify-generated-files
+	$(eval _GIT_GENERATED_FILES := $(shell echo "$(GENERATED_FILES)" | xargs git ls-files --))
+	$(eval _GIT_GENERATED_FILE_COUNT := $(words $(_GIT_GENERATED_FILES)))
+	@if [ -n "$(GENERATED_FILES)" ]; then \
```

I believe this `if` will alway evaluate to `true` because the `$(GENERATED_FILES) +=` line at the top of the file adds whitespace and `-n` specifically checks for a zero-length string.

I don't think we should try to combat whitespace in make variables, since that's just how `+=` works. Perhaps instead we could just use the count that's already computed. 

```suggestion
	@if [ $(_GIT_GENERATED_FILE_COUNT) -ne 0 ]; then \
```

Even better would be to perform the condition in `make` instead of `sh`, but I'm not sure how much of a PITA this is given the need for lazy evaluation.